sdk/python/feast/data_source.py

```diff
@@ -50,15 +50,15 @@ class KafkaOptions:
 
     def __init__(
         self,
-        bootstrap_servers: str,
+        kafka_bootstrap_servers: str,
         message_format: StreamFormat,
         topic: str,
-        watermark: Optional[timedelta] = None,
+        watermark_delay_threshold: Optional[timedelta] = None,
```

Comment:
Do we need to worry about backwards compatibility?

Reply:
no for `watermark_delay_threshold` as this was very recently added by Kevin and has not been part of a release yet

yes for `kafka_bootstrap_servers`, which I believe I handled appropriately